### DISCUSSION TITLE
Selecting word under cursor for custom key mappings?

### DISCUSSION THREAD

**Discussion by wingy3181** [JetBrains/ideavim#801] [Q&A] [answered]
Would anyone know how to copy and paste the word under the cursor for a custom key mapping?

I have the following in my .vimrc which works perfectly fine in vim but not in ideaVim

```
" [,*  ] Search and replace the word under the cursor.

nmap <leader>* :%s/\<<c-r><c-w>\>//<left>
```

It runs in ideaVim but is unable to copy and paste the word under the cursor and just appears as
`%s/\<\>//g`

I did notice this other discussion and tried using `viw` instead..........still didn't work

**Comment by lippfi** [comment]
`<C-W>` in the command line is indeed not supported. However, `viw` should work just fine. Could you please tell me more about the way you tried to use `viw`?

**Reply by wingy3181** [reply]
I must have the syntax wrong then in the rc file

I tried something like below

```
nmap <leader>* :%s/\<:viw\>//g<left><left>
nmap <leader>* :%s/\<viw\>//g<left><left>
nmap <leader>* :%s/\<<viw>\>//g<left><left>
```

but it does execute it within the command line mode and i just end up seeing the following:
<img width="312" alt="image" src="https://github.com/JetBrains/ideavim/assets/1212576/1fa7bbac-f1d6-4e5a-b2fe-2e57b8b63975">

My expectation is that it would copy the word under cursor to the command line mode

**Reply by wingy3181** [reply, marked as answer]
Ok I got what i wanted by doing this

```
nmap <leader>* viwy \| :%s/\<<c-r>+\>//g<left><left>
```

Thanks @lippfi 
Let me know if there is some other or better way i could do the same thing

**Reply by sharpchen** [reply]
> `nmap <leader>* viwy \| :%s/\<<c-r>+\>//g<left><left>`

I guess the author remapped `y` to use system clipboard, so this won't work if you have default settings.

Use `nmap <leader>s :norm! viwy<CR> :%s/\<<C-r>"\>//g<left><left>` to make sure it uses internal clipboard if you like.
